qcodes/instrument/channel.py

```diff
@@ -529,3 +532,183 @@ def validate(self, value, context: str='') -> None:
+        self._create()
+        self._exists_on_instrument = True
+
+    def _create(self) ->None:
+        """
+        (SCPI) commands needed to create the instrument. Note that we
+        need to use `self.root_instrument.write` to send commands,
+        because self.write will cause _assert_existence to raise a
+        runtime error.
+        """
+        raise NotImplementedError("Please subclass")
+
+    def delete(self) ->None:
+        """
+        Delete the channel from the instrument and remove from channel list
+        """
+        self._delete()
+        if self._channel_list is not None:
+            self._channel_list.remove(self)
+        self._exists_on_instrument = False
+
+    def _delete(self) ->None:
+        raise NotImplementedError("Please subclass")
+
+    def _assert_existence(self) ->None:
+        if not self._exists_on_instrument:
+            raise RuntimeError(
+                "Object does not exist (anymore) on the instrument")
+
+    def write(self, cmd: str) ->Any:
+        """
+        Only write to the instrument if the channel is present on the instrument
+        """
+        self._assert_existence()
+        return super().write(cmd)
+
+    def ask(self, cmd: str) ->None:
+        """
+        Only query to the instrument if the channel is present on the instrument
+        """
+        self._assert_existence()
+        return super().ask(cmd)
+
+    @property
+    def exists_on_instrument(self):
+        return self._exists_on_instrument
+
+
+class AutoLoadableChannelList(ChannelList):
+    """
+    Extends the QCoDeS ChannelList class to add the following features:
+    - Automatically create channel objects on initialization
+    - Make a `add` method to create channel objects
+    """
+    def __init__(
+            self,
+            parent: Instrument,
+            name: str,
+            chan_type: type,
+            chan_list: Optional[Sequence['AutoLoadableInstrumentChannel']]=None,
+            snapshotable: bool=True,
+            multichan_paramclass: type=MultiChannelInstrumentParameter,
+            **kwargs
+    ) ->None:
+
+        super().__init__(
+            parent, name, chan_type, chan_list, snapshotable,
+            multichan_paramclass
+        )
+
+        new_channels = self._chan_type.load_from_instrument(
+            self._parent, channel_list=self, **kwargs)
+
+        for channel in new_channels:
+            # NB: There is a bug in `extend`. TODO: Make a PR!
+            self.append(channel)
+
+    def add(self, **kwargs) ->'AutoLoadableInstrumentChannel':
+        """
+        Add a channel to the list
+        """
+        new_kwargs = self._chan_type.get_new_instance_kwargs(self._parent, **kwargs)
+
+        new_channel: AutoLoadableInstrumentChannel = self._chan_type(
+            parent=self._parent, channel_list=self, **new_kwargs
+        )
+
+        new_channel.create()
+        self.append(new_channel)
+        return new_channel
```

Comment:
are dropping the intention to rename `channel` to `submodule`? if we do the renaming, it will be important to also rename related variables/attributes in the code...

Reply:
It's not that I have dropped the intention, but rather I think this should be a PR on its own as it is a big change. Would you not agree?

Reply:
If Iapply the new naming convention partially, I fear that this will cause confusion. Since we are calling it `channel` for now, lets keep that for now.

Reply:
exactly, no need for renaming at the moment.